File: EPIJudge/epi_judge_java/epi/BuyAndSellStock.java
package epi;
import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

import java.util.List;
public class BuyAndSellStock {
  @EpiTest(testDataFile = "buy_and_sell_stock.tsv")
  public static double computeMaxProfit(List<Double> prices) {
    // TODO - you fill in here.
    double currBuy = prices.get(0), lowestBuy = prices.get(0), maxDiff = 0.0;

    for(Double currSell: prices){
      lowestBuy = Math.min(currSell, lowestBuy);
      maxDiff = Math.max(currSell-currBuy, maxDiff);
      if( maxDiff < (currSell-lowestBuy) ) currBuy=lowestBuy;
      maxDiff = Math.max(currSell-currBuy, maxDiff);
    }
    return maxDiff;
  }

  public static void main(String[] args) {
    System.exit(
        GenericTest
            .runFromAnnotations(args, "BuyAndSellStock.java",
                                new Object() {}.getClass().getEnclosingClass())
            .ordinal());
  }
}

File: EPIJudge/epi_judge_cpp/buy_and_sell_stock.cc
#include <vector>

#include "test_framework/generic_test.h"
using std::vector;

/*
 * 
 * */
double BuyAndSellStockOnce(const vector<double>& prices) {
  double max_profit = 0.0;
  double min_price = prices[0];
  for (int i = 1; i < prices.size(); ++i){
    max_profit = std::max(max_profit, prices[i] - min_price);
    min_price = std::min(min_price, prices[i]);
  }
  return max_profit;
}

int main(int argc, char* argv[]) {
  std::vector<std::string> args{argv + 1, argv + argc};
  std::vector<std::string> param_names{"prices"};
  return GenericTestMain(args, "buy_and_sell_stock.cc",
                         "buy_and_sell_stock.tsv", &BuyAndSellStockOnce,
                         DefaultComparator{}, param_names);
}
